src/utils.jl

```diff
@@ -51,14 +51,19 @@ checking if `expr` has finished executing (short for polling interval).
 """
 macro timeout(t, expr, then, pollint=0.01)
     return quote
-        tm = Float64($(esc(t)))
-        start = time()
-        tsk = @async $(esc(expr))
-        while !istaskdone(tsk) && (time() - start < tm)
-            sleep($pollint)
+        if $(esc(t)) == Inf
+            $(esc(expr))
+        else
+            tm = Float64($(esc(t)))
+            start = time()
+            tsk = @async $(esc(expr))
+            yield()
```

Comment:
Why the `yield` here?

Reply:
Per comment in #113, my concern is that `istaskdone` will return `false` and cause a 10ms sleep, even if `expr` would have executed in a very short time. The `yield()` gives the `expr` task a chance to run before checking `istaskdone`.
e.g.
```julia
function f()
       t = @async sum([1,2,3])
       !istaskdone(t) && sleep(0.01)
end
julia> f()
julia> @time f()
  0.013942 seconds (12 allocations: 1.031 KiB)
```
vs
```julia
function f()
       t = @async sum([1,2,3])
       yield()
       !istaskdone(t) && sleep(0.01)
end
julia> f()
julia> @time f()
0.000011 seconds (8 allocations: 880 bytes)
```